internal/formats/syftjson/model/package.go

```diff
@@ -124,6 +124,11 @@ func (p *Package) UnmarshalJSON(b []byte) error {
 			return err
 		}
 		p.Metadata = payload
+	case pkg.GolangBinMetadataType:
```

Comment:
There's one test that flexes each of the types, or at least most of them: https://github.com/anchore/syft/blob/main/test/integration/encode_decode_cycle_test.go I wonder if we need to add a go binary as one of the options; I think it would be mostly as simple as adding a small go binary here?: https://github.com/anchore/syft/tree/main/test/integration/test-fixtures/image-pkg-coverage/pkgs

Reply:
For flexible reproducibility (where anyone can recreate the binaries quickly, or adapt them if necessary), we can make the binaries on the fly, like we do here https://github.com/anchore/syft/tree/main/syft/pkg/cataloger/golang/test-fixtures/archs

Reply:
Yea - if we add one here I think we do it as a package `main` file and then compile the program pre-test. We then use the temp file as input and clean it up after. This way we don't commit binaries into the repo since we have ambitions on getting the OSS scorecard numbers up. I can do that as a separate PR to this one as soon as I have extra bandwidth.

Reply:
Yeah, agree not a fan of committing binaries and I know there was talk about moving to Git LFS, I think the unit test is good enough for now 👍

Reply:
We could curl a go binary in https://github.com/anchore/syft/blob/main/test/integration/test-fixtures/image-pkg-coverage/Dockerfile so we don't have to commit one. Maybe a multistage image, where we use an older syft image and pull the binary?